.github/workflows/ci.yaml

```diff
@@ -283,3 +132,266 @@ jobs:
+        with:
+          images: golemservices/golem-worker-executor
+
+      # Build and push worker executor image
+      - name: Build and push worker executor image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./golem-worker-executor-oss/docker/Dockerfile
+          push: ${{ steps.push_decision.outputs.push }}
+          platforms: linux/amd64
+          tags: ${{ steps.meta-worker-executor.outputs.tags }}
+          labels: ${{ steps.meta-worker-executor.outputs.labels }}
+
+
+      # Extract metadata (tags, labels) for shard manager
+      - name: Extract metadata (tags, labels) for shard manager
+        id: meta-shard-manager
+        uses: docker/metadata-action@v5
+        with:
+          images: golemservices/golem-shard-manager
+
+      # Build and push shard manager image
+      - name: Build and push shard manager image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./golem-shard-manager/docker/Dockerfile
+          push: ${{ steps.push_decision.outputs.push }}
+          platforms: linux/amd64
+          tags: ${{ steps.meta-shard-manager.outputs.tags }}
+          labels: ${{ steps.meta-shard-manager.outputs.labels }}
+
+      # Extract metadata (tags, labels) for golem server
+      - name: Extract metadata (tags, labels) for golem server
+        id: meta-golem-server
+        uses: docker/metadata-action@v5
+        with:
+          images: golemservices/golem-server
+
+      # Build and push cloud server image
+      - name: Build and push cloud server image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./golem-cloud-server-oss/docker/Dockerfile
+          push: ${{ steps.push_decision.outputs.push }}
+          platforms: linux/amd64
+          tags: ${{ steps.meta-golem-server.outputs.tags }}
+          labels: ${{ steps.meta-golem-server.outputs.labels }}
+
+
+  docker-build-aarch64:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Rust Targets
+        run: |
+          rustup target add aarch64-unknown-linux-gnu
+
+      - name: Install gcc-aarch64-linux-gnu
+        run: |
+          sudo apt-get update
+          sudo apt-get install gcc-aarch64-linux-gnu
+
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-release-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Should have a different cache key than the amd64 build. Although the sources are the same the generated binaries are not so I think caching with the same key in both parallel jobs just invalidates te whole thing for each other.

Reply:
Let me take a look. All I focussed was splitting what existed. May be I missed something

Reply:
Just add an `-arm64` segment to this cache key in the arm64 job.